File: Algorithms/Dynamic Programing/caching.js
function addTo80(n) {
  console.log("Long process");
  return 80 + n;
}

function memoizedAddTo80() {
  const cache = {};

  return function (n) {
    if (!(n in cache)) {
      console.log("Long process");
      cache[n] = 80 + n;
    }
    return cache[n];
  };
}

const memoize = memoizedAddTo80();

console.log("1", memoize(70));
console.log("2", memoize(70));

File: Algorithms/Dynamic Programing/caching.py
def add_to_80(n):
    print("Long process")
    return 80 + n


def memoize_add_to_80():
    cache = {}

    def new(n):
        if not cache.get(n):
            print("Long process")
            cache[n] = 80 + n
        return cache[n]
    return new


memoize = memoize_add_to_80()


print(memoize(70))
print(memoize(170))
print(memoize(70))
